outputs/lumberjack/transport.go

```
@@ -0,0 +1,203 @@
+	IsConnected() bool
+}
+
+type tcpClient struct {
+	host      string
+	connected bool
+	net.Conn
+}
+
+type tlsClient struct {
+	host      string
+	port      string
+	tls       tlsConfig
+	connected bool
+	net.Conn
+}
+
+type tlsConfig struct {
+	// MinVersion contains the minimum SSL/TLS version that is acceptable.
+	// If zero, then TLS 1.0 is taken as the minimum.
+	MinVersion uint16
+
+	// Certificates contains one or more certificate chains
+	// to present to the other side of the connection.
+	// Server configurations must include at least one certificate.
+	Certificates []tls.Certificate
+
+	// RootCAs defines the set of root certificate authorities
+	// that clients use when verifying server certificates.
+	// If RootCAs is nil, TLS uses the host's root CA set.
+	RootCAs *x509.CertPool
+}
+
+var ErrNotACertificate = errors.New("File is no certificate")
+var ErrInvalidAddress = errors.New("Invaliad address")
+var ErrDNSLookupFailed = errors.New("DNS Lookup failed")
+
+var hostportRegex, _ = regexp.Compile("^(.+):([0-9]+)$")
+
+func newTcpClient(host string) (*tcpClient, error) {
+	return &tcpClient{host: host}, nil
+}
+
+func (c *tcpClient) Connect(timeout time.Duration) error {
+	if c.IsConnected() {
+		c.Close()
+	}
+
+	conn, err := net.DialTimeout("tcp", c.host, timeout)
+	if err != nil {
+		return err
+	}
+	c.Conn = conn
+	c.connected = true
+	return nil
+}
+
+func (c *tcpClient) IsConnected() bool {
+	return c.connected
+}
+
+func (c *tcpClient) Close() error {
+	err := c.Conn.Close()
+	c.connected = false
+	return err
+}
+
+func loadTLSConfig(config *TLSConfig) (*tlsConfig, error) {
+	var tlsconfig tlsConfig
+	tlsconfig.MinVersion = tls.VersionTLS10
+
+	if len(config.Certificate) > 0 && len(config.Key) > 0 {
+		cert, err := tls.LoadX509KeyPair(config.Certificate, config.Key)
+		if err != nil {
+			logp.Critical("Failed loading client certificate", err)
+			return nil, err
+		}
+		tlsconfig.Certificates = []tls.Certificate{cert}
+	}
```

I am thinking this should return an error if Certificate is set but Key is not, or vice versa. Thoughts?